packages/data-context/src/sources/RelevantRunsDataSource.ts

```diff
@@ -243,11 +245,18 @@ export class RelevantRunsDataSource {
     if (!isEqual(toCache, this.#cached)) {
       debug('Values changed')
 
+      debug('current cache: %o, new values: %o', this.#cached, toCache)
+
       //TODO is the right thing to invalidate?  Can we just invalidate the runsByCommitShas field?
       const projectSlug = await this.ctx.project.projectId()
 
       await this.ctx.cloud.invalidate('Query', 'cloudProjectBySlug', { slug: projectSlug })
 
+      // If the cache is empty, then we're just starting up. Don't send notifications
+      if (this.#cached.all.length !== 0 && this.#cached.all[0] && toCache.all[0] && !isEqual(toCache.all[0], this.#cached.all[0])) {
```

Comment:
I'm not seeing a test for this case. It's a bit hard to read so want to make sure we have good test coverage around this logic.

Reply:
I think we do actually, search for `expect(maybeSendRunNotificationStub)` in the tests - it looks like Adam just updated existing tests that already run this function to assert it's doing the correct thing. This seems reasonable to me.